firedrake/output/vtk_output.py

```diff
@@ -477,16 +477,22 @@ def _prepare_output(self, function, max_elem):
                                     shape=shape)
         else:
             raise ValueError("Unsupported shape %s" % (shape, ))
-        output = Function(V)
+        if isinstance(function, Function):
+            output = Function(V)
+        elif isinstance(function, Cofunction):
```

Comment:
```suggestion
        else:
            assert isinstance(function, Cofunction)
```
This is just my personal preference. Means that `function` can't be anything else.

Reply:
Okay. I am fine with `assert` here. We have a previous check with an error message raised.